pkg/controller/node/nodecontroller.go

```diff
@@ -551,20 +564,14 @@ func (nc *NodeController) monitorNodeStatus() error {
 			// Check eviction timeout against decisionTimestamp
 			if observedReadyCondition.Status == api.ConditionFalse &&
 				decisionTimestamp.After(nc.nodeStatusMap[node.Name].readyTransitionTimestamp.Add(nc.podEvictionTimeout)) {
-				if nc.evictPods(node) {
-					glog.V(4).Infof("Evicting pods on node %s: %v is later than %v + %v", node.Name, decisionTimestamp, nc.nodeStatusMap[node.Name].readyTransitionTimestamp, nc.podEvictionTimeout)
-				}
+				addNodeOutageTaint(nc.kubeClient, node)
```

Comment:
Thanks for explaining, I think "don't schedule new and evict existing pods before maintenance is started" should be useful in the case users have critical pods running on a node which is going to be in maintenance.